designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala

```diff
@@ -239,12 +248,23 @@ class DBProcessService(
 
   private def validateAndReverseResolve(
       entity: ScenarioWithDetailsEntity[CanonicalProcess],
-      validateAndResolve: Boolean
+      validate: FetchScenarioGraph.ValidationMode
   )(implicit user: LoggedUser): ScenarioWithDetails = {
-    if (validateAndResolve) {
-      validateAndReverseResolve(entity)
-    } else {
-      toDisplayableProcessDetailsWithoutValidation(entity)
+    validate match {
+      case FetchScenarioGraph.ValidateAndResolve(includeValidationNodeResults) =>
+        val result = validateAndReverseResolve(entity)
+        if (includeValidationNodeResults) {
```

Comment:
```
    validate match {
      case FetchScenarioGraph.ValidateAndResolve(includeValidationNodeResults) if includeValidationNodeResults =>
        validateAndReverseResolve(entity)
      case FetchScenarioGraph.ValidateAndResolve(_) =>
        val result = validateAndReverseResolve(entity)
        result.copy(json = result.json.map { validatedProcess =>
          validatedProcess.copy(validationResult =
            validatedProcess.validationResult.map(_.copy(nodeResults = Map.empty))
          )
        })
      case FetchScenarioGraph.DontValidate =>
        toDisplayableProcessDetailsWithoutValidation(entity)
    }
 ```

Reply:
IMO it only duplicates the common part of code

Reply:
IMO it looks definitely better :)

Reply:
Done